modules/core/src/smithy4s/ShapeId.scala

```diff
@@ -40,12 +40,29 @@ object ShapeId extends ShapeTag.Has[ShapeId] { self =>
     }
   }
 
-  final case class Member(shapeId: ShapeId, member: String)
-
   val id: ShapeId = ShapeId("smithy4s", "ShapeId")
   lazy val schema =
     Schema.string.refined[ShapeId](IdRef()).withId(id)
 
+  final case class Member(shapeId: ShapeId, member: String)
+
+  object Member {
+    def parse(string: String): Option[Member] = {
```

Comment:
think if you call `ShapeId.parse` with something like `smithy4s#ShapeId$member` you will get `ShapeId("smithy4s", "ShapeId$member")` which is not entirely right